src/main/kotlin/no/nav/klage/texts/domain/TextAggregateFunctions.kt

```diff
@@ -0,0 +1,257 @@
+            timestamp = timestamp,
+        )?.let { changelogEntries.add(it) }
+
+        return TextChangedEvent(
+            text = this,
+            changelogEntries = changelogEntries,
+        )
+    }
+
+    fun Text.updateUtfall(
+        newValueUtfall: Set<String>,
+        saksbehandlerident: String
+    ): TextChangedEvent {
+        val timestamp = LocalDateTime.now()
+        val changelogEntries = mutableListOf<ChangelogEntry>()
+        val oldValueUtfall = utfall
+        utfall = newValueUtfall
+        modified = timestamp
+
+        changelog(
+            saksbehandlerident = saksbehandlerident,
+            field = Field.UTFALL,
+            fromValue = oldValueUtfall.joinToString(),
+            toValue = newValueUtfall.joinToString(),
+            timestamp = timestamp,
+        )?.let { changelogEntries.add(it) }
+
+        return TextChangedEvent(
+            text = this,
+            changelogEntries = changelogEntries,
+        )
+    }
+
+    fun Text.updateEnheter(
+        newValueEnheter: Set<String>,
+        saksbehandlerident: String
+    ): TextChangedEvent {
+        val timestamp = LocalDateTime.now()
+        val changelogEntries = mutableListOf<ChangelogEntry>()
+        val oldValueEnheter = enheter
+        enheter = newValueEnheter
+        modified = timestamp
+
+        changelog(
+            saksbehandlerident = saksbehandlerident,
+            field = Field.UTFALL,
+            fromValue = oldValueEnheter.joinToString(),
+            toValue = newValueEnheter.joinToString(),
+            timestamp = timestamp,
+        )?.let { changelogEntries.add(it) }
+
+        return TextChangedEvent(
+            text = this,
+            changelogEntries = changelogEntries,
+        )
+    }
+
+    fun Text.updateSections(
+        newValueSections: Set<String>,
+        saksbehandlerident: String
+    ): TextChangedEvent {
+        val timestamp = LocalDateTime.now()
+        val changelogEntries = mutableListOf<ChangelogEntry>()
+        val oldValueSections = sections
+        sections = newValueSections
+        modified = timestamp
+
+        changelog(
+            saksbehandlerident = saksbehandlerident,
+            field = Field.UTFALL,
+            fromValue = oldValueSections.joinToString(),
+            toValue = newValueSections.joinToString(),
+            timestamp = timestamp,
+        )?.let { changelogEntries.add(it) }
+
+        return TextChangedEvent(
+            text = this,
+            changelogEntries = changelogEntries,
+        )
+    }
+
+    private fun Text.changelog(
+        saksbehandlerident: String,
+        field: Field,
+        fromValue: String?,
+        toValue: String?,
+        timestamp: LocalDateTime
+    ): ChangelogEntry? {
+        return ChangelogEntry.changelog(
+            saksbehandlerident,
+            field,
+            fromValue,
+            toValue,
+            this.id,
+            timestamp
```

Comment:
Sikkert greit å slenge på navn her så f.eks. ikke from/to kan blandes like lett.